File: 20070224/Lua/test_cycle_length.lua
require("luaunit")
require("cycle_length")

TestCycleLength = {}
	function TestCycleLength:test()
		assertEquals(1, CycleLength:getLength(1))
		assertEquals(2, CycleLength:getLength(2))
		assertEquals(8, CycleLength:getLength(3))
		assertEquals(3, CycleLength:getLength(4))
		assertEquals(6, CycleLength:getLength(5))
	end

	function TestCycleLength:testEvenNumber()
		for i = 2, 1000, 2 do
			assertEquals(CycleLength:getLength(i / 2) + 1, CycleLength:getLength(i))
		end
	end

	function TestCycleLength:testOddNumber()
		for i = 3, 1001, 2 do
			assertEquals(CycleLength:getLength(i * 3 + 1) + 1, CycleLength:getLength(i))
		end
	end

LuaUnit:run()

File: 20070224/Ruby/TestCycleLength.rb
require 'test/unit'
require 'CycleLength'

class TestCycleLength < Test::Unit::TestCase
	def setup
		@cyclelength = CycleLength.new
	end

	def test1
		assert_equal(1, @cyclelength.getLength(1))
		assert_equal(2, @cyclelength.getLength(2))
		assert_equal(8, @cyclelength.getLength(3))
		assert_equal(3, @cyclelength.getLength(4))
	end

	def test2
		for i in (2..100)
			if i % 2 == 1 then
				assert_equal(@cyclelength.getLength(i * 3 + 1) + 1,
					@cyclelength.getLength(i))
			else
				assert_equal(@cyclelength.getLength(i / 2) + 1,
					@cyclelength.getLength(i))
			end
		end
	end
end
